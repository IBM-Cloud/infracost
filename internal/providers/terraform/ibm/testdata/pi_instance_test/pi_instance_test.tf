terraform {
  required_providers {
    ibm = {
      source  = "IBM-Cloud/ibm"
      version = "~> 1.40.0"
    }
  }
}

provider "ibm" {
  region           = "us-south"
  zone             = "dal12"
  ibmcloud_api_key = "AXrURglaTG9MApQXfqmSoME0dkcBAyv1v9Hw5vljab8y"
}

resource "ibm_pi_image" "powerimages" {
  pi_image_name        = "7200-05-03"
  pi_cloud_instance_id = "51e1879c-bcbe-4ee1-a008-49cdba0eaf60"
  pi_image_bucket_name = "images-public-bucket"
  pi_image_bucket_access = "public"
  pi_image_bucket_region = "us-south"
  pi_image_bucket_file_name = "rhcos-48-07222021.ova.gz"
  pi_image_storage_type = "tier1"
}
resource "ibm_pi_key" "key" {
  pi_key_name          = "testkey"
  pi_ssh_key           = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCKVmnMOlHKcZK8tpt3MP1lqOLAcqcJzhsvJcjscgVERRN7/9484SOBJ3HSKxxNG5JN8owAjy5f9yYwcUg+JaUVuytn5Pv3aeYROHGGg+5G346xaq3DAwX6Y5ykr2fvjObgncQBnuU5KHWCECO/4h8uWuwh/kfniXPVjFToc+gnkqA+3RKpAecZhFXwfalQ9mMuYGFxn+fwn8cYEApsJbsEmb0iJwPiZ5hjFC8wREuiTlhPHDgkBLOiycd20op2nXzDbHfCHInquEe/gYxEitALONxm0swBOwJZwlTDOB7C6y2dzlrtxr1L59m7pCkWI4EtTRLvleehBoj3u7jB4usR"
  pi_cloud_instance_id = "51e1879c-bcbe-4ee1-a008-49cdba0eaf60"
}

resource "ibm_pi_instance" "test-instance" {
  pi_memory            = "4"
  pi_processors        = "0.25"
  pi_instance_name     = "test-vm"
  pi_proc_type         = "shared"
  pi_image_id          = ibm_pi_image.powerimages.id
  pi_key_pair_name     = ibm_pi_key.key.key_id
  pi_sys_type          = "s922"
  pi_cloud_instance_id = "51e1879c-bcbe-4ee1-a008-49cdba0eaf60"
  pi_pin_policy        = "none"
  pi_health_status     = "WARNING"
  pi_storage_type  = ibm_pi_image.powerimages.pi_image_storage_type
  pi_network {
    network_id = "test-id"
  }
}
