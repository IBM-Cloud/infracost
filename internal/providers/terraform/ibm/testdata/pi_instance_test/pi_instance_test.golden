
 Name                                                                     Monthly Qty  Unit                    Monthly Cost 
                                                                                                                            
 ibm_pi_instance.aix-shared-s922-instance                                                                                   
 ├─ Storage - tier1                                                            14,600  GB hours                       $4.00 
 ├─ Cores                                                                         730  Core hours                    $92.23 
 ├─ Memory                                                                        730  GB hours                       $8.70 
 └─ Operating System                                                              730  Core hours                    $28.98 
                                                                                                                            
 ibm_pi_instance.hana-dedicated-e980-instance                                                                               
 ├─ Storage - tier1                                                            14,600  GB hours                       $4.00 
 ├─ Linux HANA Memory                                                          93,440  Memory hours                 $289.66 
 ├─ Linux HANA Cores                                                            2,920  Core hours                 $1,279.54 
 └─ Linux HANA OS for SLES (first 1 Instance hours)                                 1  Instance hours                 $0.34 
 └─ Linux HANA OS for SLES (next 2 Instance hours)                                  2  Instance hours                 $1.36 
 └─ Linux HANA OS for SLES (over 3 Instance hours)                                  2  Instance hours                 $1.64 
                                                                                                                            
 ibm_pi_instance.hana-dedicated-e980-instance_no_quantity                                                                   
 ├─ Storage - tier1                                                 Monthly cost depends on usage: $0.000274 per GB hours   
 ├─ Linux HANA Memory                                               Monthly cost depends on usage: $0.0031 per Memory hours 
 ├─ Linux HANA Cores                                                Monthly cost depends on usage: $0.44 per Core hours     
 └─ Linux HANA OS for SLES (first 1 Instance hours)                 Monthly cost depends on usage: $0.34 per Instance hours 
 └─ Linux HANA OS for SLES (next 2 Instance hours)                  Monthly cost depends on usage: $0.68 per Instance hours 
 └─ Linux HANA OS for SLES (over 3 Instance hours)                  Monthly cost depends on usage: $0.82 per Instance hours 
                                                                                                                            
 ibm_pi_instance.ibmi-dedicated-e980-instance                                                                               
 ├─ Storage - tier1                                                            14,600  GB hours                       $4.00 
 ├─ Cores                                                                         730  Core hours                 $1,291.28 
 ├─ Memory                                                                        730  GB hours                       $8.70 
 ├─ Operating System IBMi LPP                                                     730  Core hours                    $33.00 
 ├─ Operating System IBMi OS                                                      730  Core hours                 $2,067.00 
 └─ Operating System IBMi Service Extension                                       730  Core hours                   $817.02 
                                                                                                                            
 ibm_pi_instance.ibmi-dedicated-e980-instance-no-quantity                                                                   
 ├─ Storage - tier1                                                 Monthly cost depends on usage: $0.000274 per GB hours   
 ├─ Cores                                                           Monthly cost depends on usage: $1.77 per Core hours     
 ├─ Memory                                                          Monthly cost depends on usage: $0.011913 per GB hours   
 ├─ Operating System IBMi LPP                                       Monthly cost depends on usage: $0.0452 per Core hours   
 ├─ Operating System IBMi OS                                        Monthly cost depends on usage: $2.83 per Core hours     
 ├─ Cloud Storage Solution                                          Monthly cost depends on usage: $0.0959 per Core hours   
 ├─ High Availability                                               Monthly cost depends on usage: $0.30 per Core hours     
 ├─ IBM DB2 Web Query                                               Monthly cost depends on usage: $1.46 per Core hours     
 ├─ Rational Dev Studio                                             Monthly cost depends on usage: $0.16 per Instance hours 
 └─ Operating System IBMi Service Extension                         Monthly cost depends on usage: $1.12 per Core hours     
                                                                                                                            
 ibm_pi_instance.netweaver-shared-s922-instance                                                                             
 ├─ Storage - tier1                                                            14,600  GB hours                       $4.00 
 ├─ Cores                                                                         730  Core hours                    $92.23 
 ├─ Memory                                                                        730  GB hours                       $8.70 
 └─ Linux NETWEAVER OS for SLES (first 1 Instance hours)                            1  Instance hours                 $0.34 
 └─ Linux NETWEAVER OS for SLES (next 2 Instance hours)                             2  Instance hours                 $1.36 
 └─ Linux NETWEAVER OS for SLES (over 3 Instance hours)                             2  Instance hours                 $1.64 
                                                                                                                            
 ibm_pi_instance.netweaver-shared-s922-no-usage-specified-instance                                                          
 ├─ Storage - tier1                                                 Monthly cost depends on usage: $0.000274 per GB hours   
 ├─ Cores                                                           Monthly cost depends on usage: $0.13 per Core hours     
 ├─ Memory                                                          Monthly cost depends on usage: $0.011913 per GB hours   
 └─ Linux NETWEAVER OS for SLES (first 1 Instance hours)            Monthly cost depends on usage: $0.34 per Instance hours 
 └─ Linux NETWEAVER OS for SLES (next 2 Instance hours)             Monthly cost depends on usage: $0.68 per Instance hours 
 └─ Linux NETWEAVER OS for SLES (over 3 Instance hours)             Monthly cost depends on usage: $0.82 per Instance hours 
                                                                                                                            
 ibm_resource_instance.powervs_service                                                                                      
 └─ Workspace for Power Virtual Server                                              1  Workspace                      $0.00 
                                                                                                                            
 OVERALL TOTAL                                                                                                    $6,039.70 
──────────────────────────────────
14 cloud resources were detected:
∙ 8 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file
∙ 6 were free:
  ∙ 4 x ibm_pi_image
  ∙ 1 x ibm_pi_key
  ∙ 1 x ibm_resource_group